packages/esm-patient-chart-app/src/clinical-views/encounter-tile/encounter-tile.component.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useMemo } from 'react';
+import { useLayoutType } from '@openmrs/esm-framework';
+import { CodeSnippetSkeleton, Tile, Column, Layer } from '@carbon/react';
+import styles from './tile.scss';
+import { groupColumnsByEncounterType } from '../utils/helpers';
+import { useTranslation } from 'react-i18next';
+import { type EncounterTileColumn, type EncounterTileProps } from '../types';
+import { useLastEncounter } from '../hooks';
+
+export const EncounterTile = React.memo(({ patientUuid, columns, headerTitle }: EncounterTileProps) => {
+  const columnsByEncounterType = useMemo(() => groupColumnsByEncounterType(columns), [columns]);
+  const isTablet = useLayoutType() === 'tablet';
+
+  return (
+    <Layer className={styles.layer}>
+      <Tile className={styles.tile}>
+        <div className={isTablet ? styles.tabletHeading : styles.desktopHeading}>
+          <h4 className={styles.title}>{headerTitle}</h4>
+        </div>
+        <Column className={styles.columnContainer}>
+          {Object.entries(columnsByEncounterType).map(([encounterUuid, columns]) => (
+            <EncounterData
+              key={encounterUuid}
+              patientUuid={patientUuid}
+              encounterType={encounterUuid}
+              columns={columns}
+            />
+          ))}
+        </Column>
+      </Tile>
+    </Layer>
+  );
+});
+
+const EncounterData: React.FC<{
+  patientUuid: string;
+  encounterType: string;
+  columns: EncounterTileColumn[];
+}> = ({ patientUuid, encounterType, columns }) => {
+  const { lastEncounter, isLoading, error, isValidating } = useLastEncounter(patientUuid, encounterType);
+  const { t } = useTranslation();
+
+  if (isLoading || isValidating) {
+    return <CodeSnippetSkeleton type="multi" role="progressbar" />;
+  }
+
+  if (error || lastEncounter == undefined) {
+    return (
+      <div className={styles.tileBox}>
+        {columns.map((column, ind) => (
+          <div key={ind}>
+            <span className={styles.tileTitle}>{t(column.title)}</span>
+            <span className={styles.tileValue}>{error?.message}</span>
+          </div>
+        ))}
+      </div>
+    );
+  }
+
+  return (
+    <div className={styles.tileBox}>
+      {columns.map((column, ind) => {
+        return (
+          <div key={ind}>
+            <span className={styles.tileTitle}>{column.header}</span>
+            <span className={styles.tileValue}>
+              <p>{column.getObsValue(lastEncounter)}</p>
+            </span>
+            {column.hasSummary && (
+              <>
+                <span className={styles.tileValue}>
+                  <p>{column.getSummaryObsValue(lastEncounter)}</p>
+                </span>
+                <span className={styles.tileValue}>
+                  <p>{column.getObsValue(lastEncounter)}</p>
+                </span>
```

Comment:
Why do we render the current value a second time here?

Reply:
The fist block displays the actual concept answer, but in some instances, if a user wants to diaply additional information captured in a different concept, they can do it ussing a summary concept like :
```
  {
      "id": "currentRegimen",
      "hasSummary": true,
      "title": "currentRegimenTitle",
      "encounterType": "74bf4fe6-8fdb-4228-be39-680a93a9cf6d",
      "concept": "dfbe256e-30ba-4033-837a-2e8477f2e7cd",
      "summaryConcept": {
        "primaryConcept": "164515AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
      }
 },
```